agent/connect/ca/provider_vault_auth_azure.go

```diff
@@ -0,0 +1,153 @@
+package ca
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/vault/sdk/helper/jsonutil"
```

Comment:
Any particular reason for using vault's `jsonutil`? Consul has its own `DecodeJSON` too I believe which is a small wrapper around `json` from stdlib.

Reply:
I had based this off code from the vault agent which used that. We already included that dependency and I figured, as we were hitting a Vault endpoint, it made sense to keep using it.